docs/tags_doc.py

```diff
@@ -0,0 +1,112 @@
+"""
+Parses required places in the cp-ansible project for all instances of tags in following format
+tags: <test-tag>            #pattern1
+tags:                       #pattern2
+    - <test-tag>
+tags: [<tag1>, <tag2>]      #pattern3
+
+And writes the list of tags to a formatted markdown file.
+Adds new entries for new tags with blank description (to be filled by the user).
+For the existing entries with description, lets it stay as is.
+"""
+
+import os
+import re
+
+
+# Parse all tags in cp-ansible
+def get_all_tags(directory, search_text, filePattern):
+
+    all_tags = set()
+    # iterate over all yml files
+    for path, dirs, files in os.walk(os.path.abspath(directory)):
+        for filename in files:
+            if not filename.endswith(filePattern):
+                continue
+            filepath = os.path.join(path, filename)
+            with open(filepath, "r") as f:
+                file_content = f.read()
+            if search_text not in file_content:
+                continue
+            list_lines = file_content.split("\n")
+            i = 0
+            # iterate over all lines of the file
+            while (i < len(list_lines)):
+                line = list_lines[i]
+                i = i + 1
+                if len(line) == 0 or search_text not in line:
+                    continue
+                first_non_space = len(line) - len(line.lstrip())
+                if line[first_non_space] == '#':
+                    continue
+                tag = line.replace(search_text, '').strip()
+                # pattern2
+                if (tag == ''):
+                    while(i < len(list_lines) and "-" in list_lines[i]):
+                        new_tag = list_lines[i].replace("-", '').strip()
+                        if (len(new_tag.split()) == 1):
```

Comment:
This wont cover the complete use case. Below are possible and valid tag representations: 

```
      tags:
        - tag1, tag3
        - tag2
        - [tag4, tag5]
        - "tag6"
```

```
  play #1 (localhost): Host Prerequisites       TAGS: []
      TASK TAGS: [tag1, tag3, tag2, tag4, tag5, tag6]
```
please note the string sanitisation for quotes and spaces